### ISSUE TITLE
Trial Design/Conduct Interface Development - Second Iteration

### ISSUE THREAD

**Issue by eduardo-vie** [xlai/dosefinderhub#19]
This issue refers to the next steps in the development of the app's Main UI ("Trial Design/Conduct Interface").

Previously accomplished tasks can be referred to in #10.

Key upcoming tasks include:
- Set up the receiving of Sian's 3 data frames (ranking data frame, non-design-specific data frame, design-specific data frame).
- The ranking data frame will be used to extract the ranking vector, which will be used to determine the dynamic structure of the entire UI. This should relieve me from defining the mock ranking vector myself.
- The non-design-specific and design-specific data frames, which will contain info from the question database specifying UI inputs, will be used as arguments for the renderUI() function to render all Configurations tab columns (with the exception of simulation-related inputs, originally collected by me in a reactive table). This should relieve me from defining mock variables and UI inputs myself, significantly streamlining my code.
- _Related to above point:_ add a fixed non-design-specific column to the Configurations tab.
- Based on Jake's simulation outputs, plan (perhaps sketch?) and code the single Design tab, i.e., the simulation outputs tab. This will require finding user-friendly ways to allow user to visualize and compare 5 different outputs/metrics/statistics assessing design quality, for multiple simulation scenarios. This will also requiring setting up the relaying of my own data frames (ranking, non-specific, specific) to Jake, in case user updates some of the Questionnaire info in the Configuration tab.
- Based on Jake's planned outputs, plan (perhaps sketch?) and code the Conduct tab. Challenges will be similar to Design tab.
**- In summary, the main goal is to have a fully function Main UI ASAP (!!!!!), although not necessarily featuring all our intended outputs.**

**Comment by eduardo-vie** [comment]
Quick update on the Main UI:
- I have automated the Configurations tab based on the dummy data
- I have coded the Simulation tab inputs and the Conduct tab inputs (including reactive table)
(See screenshots below)

Main next steps include:
- Adding an "upload data" option for the user in the Configurations tab and setting up the dataflow from me to Jake (should be very similar to the flow from Sian to me, but adding a simulation parameters data frame)
- Get and plot outputs from Jake?
- General user-experience / aesthetic improvements
- Server function modularization

![image](https://github.com/xlai/dosefinderhub/assets/141162230/a0e5692b-e9a9-4d4d-92e0-286aa0e21011)

![image](https://github.com/xlai/dosefinderhub/assets/141162230/1c2f6ede-c294-45b9-938d-de9aba892809)

![image](https://github.com/xlai/dosefinderhub/assets/141162230/51ffb1e6-0f42-4ed0-af13-fd278bbfcf00)

**Comment by xlai** [comment]
This looks great! can't wait to see the finished product

**Comment by xlai** [comment]
I will close this issue now as some (e.g. simulation tab reactive table) were partially addressed by #44 . Improvement on conduct tab will be brought over to a separate issue once the new main app UI has been implemented.
